pkgs/tools/X11/caffeine-ng/default.nix

```diff
@@ -1,53 +1,90 @@
-{ gdk-pixbuf, glib, gobject-introspection, gtk3, lib, libnotify,
-  procps, xset, xautolock, xscreensaver, python3Packages, wrapGAppsHook
+{ buildPythonApplication
+, fetchPypi
+
+, gobject-introspection
+, gtk3
+, lib
+, libappindicator-gtk3
+, libnotify
+, setuptools-scm
+
+, click
+, dbus-python
+, ewmh
+, pulsectl
+, pygobject3
+, pyxdg
+, setproctitle
+
+, python3
+, procps
+, xset
+, xautolock
+, xscreensaver
+, xfce
+
+, glib
+, wrapGAppsHook
 }:
 
-python3Packages.buildPythonApplication rec {
+let
+  click_7 = click.overridePythonAttrs (old: rec {
+    version = "7.1.2";
+    src = old.src.override {
+      inherit version;
+      sha256 = "d2b5255c7c6349bc1bd1e59e08cd12acbbd63ce649f2588755783aa94dfb6b1a";
+    };
+  });
+in buildPythonApplication rec {
   pname = "caffeine-ng";
-  version = "3.5.1";
+  version = "4.0.2";
 
-  src = python3Packages.fetchPypi{
+  src = fetchPypi {
     inherit pname version;
-    sha256="0akzldqvxnqngpj1s6y2phgj7ch8wfm02j6z2drqvsbvaadw0jbm";
+    sha256 = "sha256-umIjXJ0et6Pi5Ejj96Q+ZhiKS+yj7bsgb4uQW6Ym6rU=";
   };
 
   nativeBuildInputs = [ wrapGAppsHook glib ];
+
   buildInputs = [
-    gdk-pixbuf gobject-introspection libnotify gtk3
-    python3Packages.setuptools-scm
+    libappindicator-gtk3
+    libnotify
+    gobject-introspection
+    gtk3
+    setuptools-scm
   ];
-  pythonPath = with python3Packages; [
-    dbus-python docopt ewmh pygobject3 pyxdg
-    setproctitle pulsectl
+
+  pythonPath = [
+    click_7
+    dbus-python
+    ewmh
+    pulsectl
+    pygobject3
+    pyxdg
+    setproctitle
   ];
 
   doCheck = false; # There are no tests.
 
-  postPatch = ''
-    substituteInPlace caffeine/inhibitors.py \
-      --replace 'os.system("xset' 'os.system("${xset}/bin/xset' \
-      --replace 'os.system("xautolock' 'os.system("${xautolock}/bin/xautolock' \
-      --replace 'os.system("pgrep' 'os.system("${procps}/bin/pgrep' \
-      --replace 'os.system("xscreensaver-command' 'os.system("${xscreensaver}/bin/xscreensaver-command'
-  '';
```

Comment:
Did you have a special reason for removing this?
On my machine, this makes e.g. `xautolock` lock the screen despite `caffeine-ng` being enabled, if `xautolock` is not in the `$PATH` for `caffeine-ng` (I haven't tested it with `xautolock` installed globally).

Reply:
I did add them in `$PATH` in `gappsWrapperArgs`.

Reply:
Ah, that's even better (and works too).